### PR TITLE
add pagination component

### PR DESCRIPTION
**Pull request by lilyanB**
figma:
image

### CODE REVIEW

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  currentPage: number;
+  customClass?: string;
+  onPageChange: (newPage: number) => void;
+}
+
+export function Pagination(props: PaginationProps) {
+  const { pages, currentPage, customClass, onPageChange } = props;
+  const pageClass = 'px-3 py-1 rounded-lg cursor-pointer';
+  const currentpageClass = `${pageClass} bg-f-primary text-f-secondary`;
+  const renderPageNumbers = (): JSX.Element[] => {
```
Comment (maxime-carabina) [inline src/components/Pagination/Pagination.tsx:19]:
```suggestion
  function renderPageNumbers() {
```
Can you move this function into a new file ?
And separate his content to multiple explicit function ?

Too complicated to understand.

Comment (lilyanB) [reply src/components/Pagination/Pagination.tsx]: done

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    pagination.push(
+      <span className={pageClass} key={key}>
+        ...
+      </span>,
+    );
+  };
+
+  addPageLink(1, currentPage === 1 ? currentpageClass : pageClass);
+
+  if (pages > 7) {
```
Comment (mazmassa) [inline src/components/Pagination/Pagination.tsx:41]:
```suggestion
const MAX_PAGES = 7;

  if (pages > MAX_PAGES) {
```

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        addPageLink(i, className);
+      }
+    } else if (currentPage < 3) {
+      for (let i = pages - 2; i < pages; i++) {
+        const className = i === currentPage ? currentpageClass : pageClass;
+        addPageLink(i, className);
+      }
+    }
+  }
+  if (pages > 1 && pages <= 7) {
```
Comment (mazmassa) [inline src/components/Pagination/Pagination.tsx:88]:
```suggestion
  if (pages > 1 && pages <= MAX_PAGES) {
```

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    if (currentPage > 2 && !isFirstPage) {
+      addEllipsis('startEllipsis');
+    }
+
+    if (isFirstPage) {
+      for (let i = currentPage + 1; i < Math.min(currentPage + 4, 4); i++) {
+        addPageLink(i, i === currentPage ? currentpageClass : pageClass);
+      }
+    } else if (isLastPage) {
+      for (let i = currentPage - 4; i < pages; i++) {
```
Comment (mazmassa) [inline src/components/Pagination/Pagination.tsx:51]: What is the 4 here!? Give a name in a const for that MIN_LEFT_PAGES, etc...

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        onClick={() =>
+          currentPage !== 1 ? onPageChange(currentPage - 1) : undefined
+        }
+        className={currentPage !== 1 ? pageClass : 'px-3 py-1 rounded-lg'}
+      >
+        <MdArrowForwardIos className="rotate-180" />
+      </a>
+
+      {pagination}
+
```
Comment (mazmassa) [inline src/components/Pagination/Pagination.tsx:135]:
```suggestion
      {pagination}
```

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { MdArrowForwardIos } from 'react-icons/md';
```
[the remark this answers is not in the document]
Comment (lilyanB) [reply src/components/Pagination/Pagination.tsx:5]: done

File: src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (lilyanB) [reply src/components/Pagination/Pagination.tsx:88]: done

File: src/components/Pagination/Pagination.tsx (hunk 4 above)
Comment (lilyanB) [reply src/components/Pagination/Pagination.tsx:51]: done

File: src/components/Pagination/Pagination.tsx (hunk 6 above)
Comment (mazmassa) [inline src/components/Pagination/Pagination.tsx:5]: I think here also you need to find the correspondent under 'react-icons/fi'.

Comment (mazmassa) [reply src/components/Pagination/Pagination.tsx:5]: Here: https://react-icons.github.io/react-icons/icons?name=fi

Comment (lilyanB) [reply src/components/Pagination/Pagination.tsx:5]: yes it's "import { FiChevronRight } from 'react-icons/fi';"

**Review by mazmassa — APPROVED** [review]
Thanks for that Lilyan! 👏 

Really nice!

**Pull request merged**